### ISSUE TITLE
Potential Bug: Default profile always Desktop and not X360?

### ISSUE THREAD

**Issue by lyndonguitar** [ayufan/steam-deck-tools#57]
Is it possible to have it start using X360 immediately? it keeps on switching back to Desktop mode at boot, even if I set the "default profile" to X360 in the settings. I have "Use X360 Controller with Steam"

It works for the most part, but I have to always switch manually to X360 mode from Desktop,  I prefer a X360 setup as default start

**EDIT:** I have traced the problem to a method that I use
**Quick explanation:** 

I am using Playnite with[ this method](https://playnite.link/forum/thread-967.html) so that Playnite fullscreen starts first before the actual Windows explorer. This way I can easily replicate a SteamOS experience. 

Everything works as intended, except this somehow breaks the detection mechanism of Steam-Deck-Tools and causes it to switch to Desktop mode at start, despite the default profile being switched to X360 and my Playnite being at fullscreen.

As soon as I restart Playnite, or manually switch it to X360, the detection fixes itself. I can also hold Steam+Options to Toggle desktop mode and that somehow fixes it and goes in to gamepad mode

Is there a way to "brute" force Controller to switch to X360? Without relying on the detection mechanism?

**Comment by ayufan** [comment]
> Is there a way to "brute" force Controller to switch to X360? Without relying on the detection mechanism?

Interesting. The default should make controller to always start with this profile instead of Desktop. Will validate it.

**Comment by ayufan** [comment]
@lyndonguitar Do you have in Settings `X360` configured? As this should be enough to make it "on by default", in all cases.

**Comment by lyndonguitar** [comment]
> @lyndonguitar Do you have in Settings `X360` configured? As this should be enough to make it "on by default", in all cases.

Yes. interestingly enough holding Steam+Option (hotkey) at start switches it to Gamepad mode.

**To further elaborate:**
I use a Playnite method wherein it becomes the temporary starting shell, to boot it before the windows explorer. So there is a seamless boot-up process that doesn't show any sign of Windows. Pretty harmless for the most part.

In steam-deck-tools. Everything is set to X360. Default Profile, even the last setting before closing is X360.

When I boot the Steam Deck, the steam-deck-tools always reverts to Desktop mode, even if the default is X360. The only way to fix this is by manually selecting it in the settings, or holding Steam+Option, followed by a USB connection sound.

Steam-deck-tools also manages to fix itself IF I restart Playnite, or interestingly enough, switch from Playnite Fullscreen to Playnite desktop mode.

Maybe Desktop mode is the de-facto Default mode for the program, and even tho I set the default to X360 mode, the unconventional Playnite setup inadvertently breaks that setting from taking effect, so it reverts to its most basic setting, which is the Desktop. I am wishing there is a way to bypass that? 

In anyway I'm conducting more tests later

**Comment by lyndonguitar** [comment]
**Important Update.** So, It's not about the shell hack that I use. Its all about Playnite opening first than steam-deck-tools
I reverted all the Shell tweaks that I made, and just ran explorer.exe. No run at startup for both Playnite and steamcontroller.exe

I run playnite Fullscreen.
Then I run steam controller
=This bug happens

I run steam controller
Then I run playnite Fullscreen.
=No bug

**Comment by ayufan** [comment]
It is still interesting, as no Desktop should be used if X360 is configured.

Would you mind running debug version from https://github.com/ayufan/steam-deck-tools/releases/tag/0.6.7, and gathering logs with https://learn.microsoft.com/en-us/sysinternals/downloads/debugview?

Once you do it, then just reinstall to release to avoid using experimental features unless you want them.

**Comment by lyndonguitar** [comment]
Okay, so I tested it with your instructions
-uninstalled steam deck tools
-installed the debug version
-downloaded debug view

Boot up windows. Prevent steamcontroller.exe and playnite from running at start, so I could open debug view first.
I opened the non 64bit debugview and did not change any settings

Ran Playnite, and then Steamcontroller.exe
=Bug appeared.
Copied Logs here 
`[10740] RecursiveDirectoryCreate( C:\Users\Administrator\AppData\Local\Playnite\User Data directory exists )
[13868] Profiler was prevented from loading notification profiler due to app settings.
[13868]   Process ID (decimal): 13868.  Message ID: [0x2509].
[9068] Profiler was prevented from loading notification profiler due to app settings.
[9068]   Process ID (decimal): 9068.  Message ID: [0x2509].
[8080] Profiler was prevented from loading notification profiler due to app settings.
[8080]   Process ID (decimal): 8080.  Message ID: [0x2509].
[12900] Profiler was prevented from loading notification profiler due to app settings.
[12900]   Process ID (decimal): 12900.  Message ID: [0x2509].
[13320] Profiler was prevented from loading notification profiler due to app settings.
[13320]   Process ID (decimal): 13320.  Message ID: [0x2509].
`

closed debug and reset everything
I did the reverse 
(Open Steamcontroller.exe first,, and at this point the x360 worked as normal, and then Playnite)
Here are the logs
`[12100] Profiler was prevented from loading notification profiler due to app settings.
[12100]   Process ID (decimal): 12100.  Message ID: [0x2509].
[984] Profiler was prevented from loading notification profiler due to app settings.
[984]   Process ID (decimal): 984.  Message ID: [0x2509].
[11796] Local Device Found
[11796]   type: 045e 028e
[11796]   path: sdl://1
[11796]   serial_number:  - 0
[11796]   Manufacturer: 
[11796]   Product:      XInput Controller
[11796]   Release:      0
[11796]   Interface:    -1
[11796] 
[11796] !! Steam controller device opened for index 0.
[11796] PNG load warning iCCP: known incorrect sRGB profile
[2748] RecursiveDirectoryCreate( C:\Users\Administrator\AppData\Local\Playnite\User Data directory exists )
`
